src/post.cc

```diff
@@ -642,11 +642,12 @@ void post_t::add_to_value(value_t& value, const optional<expr_t&>& expr) const
       add_or_set_value(value, xdata_->compound_value);
   }
   else if (expr) {
-    bind_scope_t bound_scope(*expr->get_context(),
-                             const_cast<post_t&>(*this));
+    scope_t *ctx = expr->get_context();
+    bind_scope_t bound_scope(*ctx, const_cast<post_t&>(*this));
 #if 1
     value_t temp(expr->calc(bound_scope));
     add_or_set_value(value, temp);
+    expr->set_context(ctx);
```

Comment:
* If we can copy `expr` to `tmp_expr` and modify the `tmp_expr` to `bound_scope` that'll be cleaner, but `expr_t` is reference counted pointer and maybe not that easy
* `set_context` doesn't invalidate the `compile` flag, is that ok? (I don't have deep understanding, so just a speculation)